hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/tokentypes/Token.java

```diff
@@ -0,0 +1,244 @@
+                token.autoRenewPeriod() == -1L ? Optional.empty() : Optional.of(token.autoRenewPeriod()),
+                token.accountsAreFrozenByDefault(),
+                token.accountsKycGrantedByDefault(),
+                token.treasury(),
+                token.autoRenewAccount(),
+                token.customFeeSchedule(),
+                token.adminKey(),
+                token.feeScheduleKey(),
+                token.freezeKey(),
+                token.kycKey(),
+                token.pauseKey(),
+                token.supplyKey(),
+                token.wipeKey());
+        Objects.requireNonNull(tokenRes.tokenType, "tokenType");
+        Objects.requireNonNull(tokenRes.tokenSupplyType, "tokenSupplyType");
+        Objects.requireNonNull(tokenRes.symbol, "symbol");
+        Objects.requireNonNull(tokenRes.name, "name");
+        Objects.requireNonNull(tokenRes.memo, "memo");
+        Objects.requireNonNull(tokenRes.adminKey, "adminKey");
+        Objects.requireNonNull(tokenRes.feeScheduleKey, "feeScheduleKey");
+        Objects.requireNonNull(tokenRes.freezeKey, "freezeKey");
+        Objects.requireNonNull(tokenRes.kycKey, "kycKey");
+        Objects.requireNonNull(tokenRes.pauseKey, "pauseKey");
+        Objects.requireNonNull(tokenRes.supplyKey, "supplyKey");
+        Objects.requireNonNull(tokenRes.wipeKey, "wipeKey");
+
+        return tokenRes;
+    }
+
+    static Token fromMod(@NonNull final com.hedera.hapi.node.state.token.Token token) {
+        Token tokenRes;
+        try {
+            tokenRes = new Token(
+                    TokenType.valueOf(token.tokenType().protoName()),
+                    token.supplyType(),
+                    token.tokenId().tokenNum(),
+                    token.symbol(),
+                    token.name(),
+                    token.memo(),
+                    token.deleted(),
+                    token.paused(),
+                    token.decimals(),
+                    token.maxSupply(),
+                    token.totalSupply(),
+                    token.lastUsedSerialNumber(),
+                    token.expirationSecond(),
+                    token.autoRenewSeconds() == -1L ? Optional.empty() : Optional.of(token.autoRenewSeconds()),
+                    token.accountsFrozenByDefault(),
+                    token.accountsKycGrantedByDefault(),
+                    idFromMod(token.treasuryAccountId()),
+                    idFromMod(token.autoRenewAccountId()),
+                    customFeesFromMod(token.customFees()),
+                    keyFromMod(token.adminKey()),
+                    keyFromMod(token.feeScheduleKey()),
+                    keyFromMod(token.freezeKey()),
+                    keyFromMod(token.kycKey()),
+                    keyFromMod(token.pauseKey()),
+                    keyFromMod(token.supplyKey()),
+                    keyFromMod(token.wipeKey()));
+        } catch (InvalidKeyException e) {
+            // todo check how to handle invalid key exceptions
```

Comment:
@david-bakin-sl or @tinker-michaelj thoughts on this? We could put some text like "invalid key" in the output instead of throwing an exception

Reply:
Done. Instead of throwing [here](https://github.com/hashgraph/hedera-services/pull/11543/commits/e883ea21aa09ae623e3920942be9a9af39a37486#diff-83c53a12be551e2f19efdf69998953eadf3ea7c7405b5a08504a06364c232e4fR189-R202) while gathering keys from mod, I'm returning an invalid key, that will be reported as invalid [here](https://github.com/hashgraph/hedera-services/pull/11543/commits/e883ea21aa09ae623e3920942be9a9af39a37486#diff-c4e227a3be393f13d5c3c67b28d2548a84707f74ef88a21609960c670b88e9e3R288).

Reply:
Great, ty!